docs/DeveloperGuide.adoc

```diff
@@ -458,8 +454,14 @@ image::SortAllSequenceDiagram.png[]
 
 [NOTE]
 ====
-* The lifeline for `SortCommandParser`` should end at the
+* The lifeline for `SortCommandParser` should end at the
 destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+* The `Logic` component of the `Sort` commands differs from other components as the
+`Module`, `Group` and `Student` components have their own parsers (such as the `ModuleCommandParser`,
+`GroupCommandParser` and `StudentCommandParser` respectively) which direct the commands
+towards their own individual parsers. The `Sort` commands, on the other hand, have
+a single `SortCommandParser` which handles the parsing for all three kinds of commands.
```

Comment:
The `SortCommandParser`, which creates `Sort` commands, is different from the other command parsers. While the other parser have another level of parsing, the `SortCommandParser` creates all  the different `Sort` commands within itself.